### PR TITLE
Change tagbot workflow to run on pull_request_target and comment on security

### PR DESCRIPTION
**Pull request by Micket**
See https://github.com/actions/labeler?tab=readme-ov-file#permissions

### CODE REVIEW

**Comment by Micket** [comment]
On @ocaisa 's request i reworked the workflow to print a warning if someone modifies the script. 

This will not run here yet, since the develop branch version doesn't have pull_request_target yet, so I will make a demonstration on my own repo and link here.

**Comment by Micket** [comment]
Well this rabbit hole was deep. Had to, like many others, rediscover https://github.com/actions/checkout/issues/518
1. separate checkout for the PR, so that the PR itself has no possible way of impacting the script, and can't do anything sneaky like importing something like "git.py" to indirectly modify it via the script doing import git
2. since i can't rely on merge commit sha, i manually check for a merge conflict
3. I added a workflow tag for anything that changes workflows to highlight possible changes to highlight such (potentially dangerous) changes.

Also other cleanups, tweaks, to make things more robust and nicer.

**Comment by Micket** [comment]
Example PRs on my own fork to show the bot in action:
1. Updating a software https://github.com/Micket/easybuild-easyconfigs/pull/24 (also removed a workflow file, so it gets a workflow tag)
2. Trying to modify the script has no impact (as intended) https://github.com/Micket/easybuild-easyconfigs/pull/31 and there is a workflow tag.
3. Trying to modify workflow yml directly https://github.com/Micket/easybuild-easyconfigs/pull/32 does not impact the PR as intended.

File: .github/workflows/tagbot.yml [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
+on: [pull_request_target]
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
 
 jobs:
   tagbot:
+    # Note: can't rely on github.event.pull_request.merge_commit_sha because pull_request_target
+    # does not wait github mergability check, and the value is outdated.
```
Comment (ocaisa) [inline .github/workflows/tagbot.yml:11]:
```suggestion
    # does not wait for github mergability check, and the value is outdated.
```

**Review by ocaisa — APPROVED** [review]
The proof will be in the pudding

**Comment by ocaisa** [comment]
Going in, thanks for all the effort on this @Micket !

**Pull request merged**